src/sc_client/models.py

```diff
@@ -78,10 +78,19 @@ def triple(
         param2: ScTemplateParam,
         param3: ScTemplateParam,
     ) -> None:
+        for param in (param1, param2, param3):
+            if isinstance(param, List):
+                self._is_var_type(param[0])
+            else:
+                self._is_var_type(param)
         p1, p2, p3 = tuple(map(self._split_template_param, [param1, param2, param3]))
 
         self.triple_list.append(ScTemplateTriple(src=p1, edge=p2, trg=p3))
 
+    def _is_var_type(self, param: ScTemplateParam):
+        if isinstance(param, ScType) and param.is_const():
+            raise InvalidValueError("You should to use variable types in template")
```

Comment:
It's better to name like InvalidTypeError

Reply:
Agreed, also refactor another parts of code to raise such error